Framework/Algorithms/src/GetAllEi.cpp

```diff
@@ -894,15 +897,15 @@ GetAllEi::buildWorkspaceToFit(const API::MatrixWorkspace_sptr &inputWS,
   // general.
   working_ws->setX(0, bins);
   working_ws->setX(1, bins);
-  MantidVec &Signal1 = working_ws->dataY(0);
-  MantidVec &Error1 = working_ws->dataE(0);
-  MantidVec &Signal2 = working_ws->dataY(1);
-  MantidVec &Error2 = working_ws->dataE(1);
+  auto &Signal1 = working_ws->mutableY(0);
+  auto &Error1 = working_ws->mutableE(0);
+  auto &Signal2 = working_ws->mutableY(1);
+  auto &Error2 = working_ws->mutableE(1);
   for (size_t i = 0; i < YLength; i++) {
-    Signal1[i] = inputWS->dataY(wsIndex0)[i];
-    Error1[i] = inputWS->dataE(wsIndex0)[i];
-    Signal2[i] = inputWS->dataY(wsIndex1)[i];
-    Error2[i] = inputWS->dataE(wsIndex1)[i];
+    Signal1[i] = inputWS->mutableY(wsIndex0)[i];
```

Comment:
Is it possible to use `assign` here? Also you should have been using the const reference from inputWS e.g `inputWS->y(wsIndex0)`. However, I think you can use the `assign` methods here instead.